lib/c/libintvector.h

```diff
@@ -25,6 +27,10 @@
 #include <tmmintrin.h>
 #include <smmintrin.h>
 
+#if defined(_MSC_VER)
+  #include <intrin.h> 
```

Comment:
As @msprotz mentioned, some information on this would be good. Why is this needed now and what wasn't working before?

Reply:
I think I just remembered: intrin.h breaks the build when using cygwin-packaged, native mingw compilers. We should have CI for that toolchain, too.

Reply:
Would it then be better to have smth like: 
```

#ifdef _MSC_VER
#if _MSC_VER > version
 
 #include <intrin.h> 
#endif
#endif
```

Reply:
Or will mingw be already filtered out by the `#ifdef _MSC_VER` clause?

Reply:
I'm not sure -- I think you would need to check. Basically, I'm fine with any solution, as long as it doesn't break the mingw-based setup that several of us currently rely on.

@tahina-pro do we still have windows CI anywhere in our system that could check Natalia's patch? Or alternatively, could you tell us quickly if `#include <intrin.h>` works with x86_64-mingw64-gcc?

Reply:
Right now we have no working Windows CI on our side. We only have a nightly `./everest make test` on Windows, which has been failing for months now...
Re. intrin.h with x86_64-mingw64-gcc, I managed to compile a dummy C file containing just that include, but I guess that won't tell us much. I don't fully understand here, but I think the header in question might be https://sourceforge.net/p/mingw-w64/mingw-w64/ci/master/tree/mingw-w64-headers/crt/intrin.h and a related discussion might be https://mingw-w64-public.narkive.com/9uy03D8V/what-is-the-purpose-of-intrin-h

Reply:
thanks Tahina... so `intrin.h` appears to define MSVC-like macros for mingw... at this point I'm thoroughly confused why we need that header

maybe to re-center the discussion, @frosne, you can provide us with the compilation error that this include fixes? thanks!

Reply:
It looks like it's indeed required by mingw

Reply:
I would suggest pulling this change out of this PR and handling it separately as it appears to need clarification. The `__has_include` is uncontroversial and could go in.

Looking at the [official documentation for MSVC](https://docs.microsoft.com/en-us/cpp/intrinsics/x86-intrinsics-list?view=msvc-170) `intrin.h` should be the only one needed. The others (`tmmintrin.h` and `smmintrin`) appear to be outdated.

MinGW supports `intrin.h` as well such that it should work for both environment.
We will have a Windows CI in a couple weeks to test this on and I might be able to tell you more next week. But I have to set up a Windows environment first.
If someone else has a Windows setup, they could give this a go with `intrin.h` only.